libmamba/src/core/package_handling.cpp

```diff
@@ -57,7 +58,104 @@ namespace mamba
         const fs::u8path& m_file;
     };
 
-    static int copy_data(archive* ar, archive* aw)
+    class libarchive_read_raii
```

Comment:
I don't think having `raii` in the name is useful, nor is having the `lib` prefix. Suggestion: `scoped_archive_read` (same suggestion for the other names)